sos/upload/targets/__init__.py

```diff
@@ -0,0 +1,777 @@
+# Copyright 2024 Red Hat, Inc. Jose Castillo <jcastillo@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import os
+import re
+import logging
+
+from getpass import getpass
+from sos import _sos as _
+from sos.utilities import is_executable, TIMEOUT_DEFAULT
+
+try:
+    import requests
+    REQUESTS_LOADED = True
+except ImportError:
+    REQUESTS_LOADED = False
+
+try:
+    import boto3
+    BOTO3_LOADED = True
+except ImportError:
+    BOTO3_LOADED = False
+
+
+class UploadTarget():
+    """
+    This class is designed to upload files to a distribution
+    defined location. These files can be either sos reports,
+    sos collections, or other kind of files like: vmcores,
+    application cores, logs, etc.
+
+    """
+
+    desc = """
+            Upload a file (can be a sos report, a must-gather, or others) to
```

Comment:
Super minor nit that I'm only highlighting since this string can be user facing: this should be "an sos report", as it is officially "ess oh ess" and not "sauce".